server/api/impl/proxy/proxy.go

```diff
@@ -19,18 +19,26 @@ import (
 )
 
 type Proxy struct {
-	builtIn map[apps.AppID]api.Upstream
+	builtinUpstreams map[apps.AppID]api.Upstream
 
-	mm        *pluginapi.Client
-	conf      api.Configurator
-	store     api.Store
-	awsClient *aws.Client
+	mm            *pluginapi.Client
+	conf          api.Configurator
+	store         api.Store
+	aws           aws.Service
+	s3AssetBucket string
```

Comment:
Why do we store bucket name here and not in the aws?

Reply:
1/5 By keeping configuration issues out of the client, we can define how we configure each use case's bucket where it needs to be? The Client is now as simple as it can be, just a convenience wrapper on top of the AWS API + auth, retries, etc.